pkg/apis/hive/v1/baremetal/platform.go

```diff
@@ -1,5 +1,12 @@
 package baremetal
 
+import corev1 "k8s.io/api/core/v1"
+
 // Platform stores the global configuration for the cluster.
 type Platform struct {
+	// LibvirtSSHPrivateKeySecretRef is the reference to the secret that contains the private SSH key to use
+	// for access to the libvirt provisioning host.
+	// The SSH private key is expected to be in the secret data under the "ssh-privatekey" key.
+	// +optional
+	LibvirtSSHPrivateKeySecretRef *corev1.LocalObjectReference `json:"libvirtSSHPrivateKeySecretRef,omitempty"`
```

Comment:
Is this truly optional? Is this to cover the case where the ssh key provided in the `.spec.provisioning.sshPrivateKeySecretRef` is the same key used for libvirt? If so, I think that I would prefer that the user still explicitly specified the ssh key here.

Reply:
It's optional only because I don't think it will be required long term. The whole libvirt host to provision thing should go away someday so it seemed best to make this optional. If we can easily make it mandatory now and optional later then that would be fine, but initially I wouldn't think it ideal to change things back and forth in the API like that.

Reply:
Hmm, should we leave it optional and validate that it's set today?

Reply:
I think that it is easy to go from mandatory to optional but not the other way around. So if we start with optional, then it will always be optional.

Reply:
Ok updated to make required.